AsyncDisplayKit/Private/ASDisplayNode+UIViewBridge.mm

```diff
@@ -214,9 +234,11 @@ - (CGRect)frame
 
 - (void)setFrame:(CGRect)rect
 {
-  _bridge_prologue;
-
-  if (_flags.synchronous && !_flags.layerBacked) {
+  _bridge_prologue_write;
+  BOOL setFrameDirectly = _flags.synchronous && !_flags.layerBacked;
+  BOOL isMainThread = ASDisplayNodeThreadIsMain();
+  BOOL nodeLoaded = self.nodeLoaded;
```

Comment:
@appleguy Hmm this is pretty interesting.
- The `__loaded` macro in this file evaluates to `_layer != nil`. As far as I can tell, this is incorrect because `_layer` will never bet set on view-backed nodes. (`__loadNode` -> `[self layer]` -> `return self.view.layer` short-circuit.)
- `nodeLoaded` isn't backed by an instance variable – it's computed as a combination of `_view` `_layer` and `_flags.layerBacked`.

How should I proceed? Is there some reason the `__loaded` macro behaves this way? It's used all over this file so we need it to be correct. The lightest-weight solution here is to replace the short-circuit I mentioned above in `-layer` with `_layer = self.view.layer; return _layer` so that after `__loadNode` we guarantee that `_layer` is non-nil.

For now what I'll do is apply the same logic that that method uses here.